src/controllers/registrationController.js

```diff
@@ -0,0 +1,107 @@
+const mongoose = require('mongoose');
+const Registration = require('../models/registration');
+const cache = require('../utilities/nodeCache')();
+const logger = require('../startup/logger');
+
+const registrationController = function () {
+    const createRegistration = async function (req, res) { 
+
+      if (!req.body.eventId) {
+        return res.status(400).send({
+          error: 'Event ID is a mandatory field'
+        });
+      }
+  
+      try {
+        const { eventId } = req.body;
+        const userId = req.body.requestor.requestorId;
```

Comment:
```suggestion
        const { eventId } = req.body.eventId?.trim();
        const userId = req.body.requestor?.requestorId?.trim();
```

Recommend trimming eventId and requestorId using .trim() to ensure no unintended leading or trailing spaces.

Reply:
Thank you for the review and suggestion. However, in this case, trimming might not be necessary since both eventId/registrationId and userId are MongoDB ObjectIds that come from our system (database and authentication) rather than direct user input. The existing MongoDB ObjectId validation `mongoose.Types.ObjectId.isValid()` will catch any malformed IDs. If we do need to handle string cleaning in the future, I think it would be better handled at the input validation layer or form processing level prior to passing into the controller.